src/net_processing.cpp

```diff
@@ -3855,15 +3867,19 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             }
 
             if (pindex->nHeight >= m_chainman.ActiveChain().Height() - MAX_BLOCKTXN_DEPTH) {
-                CBlock block;
-                bool ret = ReadBlockFromDisk(block, pindex, m_chainparams.GetConsensus());
-                assert(ret);
-
-                SendBlockTransactions(pfrom, *peer, block, req);
-                return;
+                block_pos = pindex->GetBlockPos();
             }
         }
 
+        if (!block_pos.IsNull()) {
+            CBlock block;
+            bool ret = ReadBlockFromDisk(block, block_pos, m_chainparams.GetConsensus());
+            assert(ret);
```

Comment:
What about this assertion? Couldn't `ReadBlockFromDisk` return false if the block is not on disk anymore?

Reply:
See https://github.com/bitcoin/bitcoin/pull/26326#discussion_r1043171470. Blocks below `MAX_BLOCKTXN_DEPTH` can't be pruned, so we should retain the current behavior if this read fails.